src/quacc/recipes/espresso/bands.py

```diff
@@ -0,0 +1,284 @@
+) -> BandsSchema:
+    """
+    Function to compute bands structure and fermi surface using pw.x, bands.x and fs.x.
+    This is all done in a single job as a multi-step process.
+
+    1. A pw.x non-self consistent calculation
+        - name: "bands"
+
+    2. A bands.x post-processing calculation
+        - name: "bands_pp"
+
+    3. A fs.x calculation to obtain the fermi surface
+        - name: "fermi_surface"
+
+    Parameters
+    ----------
+    atoms
+        The Atoms object.
+    prev_dir
+        Outdir of the previously ran pw.x calculation. This is used to copy
+        the entire tree structure of that directory to the working directory
+        of this calculation.
+    run_bands_pp
+        If True, a bands.x post-processing calculation will be carried out.
+        This allows to re-order bands and computes band-related properties.
+    run_fermi_surface
+        If True, a fs.x calculation will be carried out.
+        This allows to generate the fermi surface of your structure.
+    primitive
+        If True, it gets the primitive cell for your structure and generate
+        the high symmetry k-path using Latimer-Munro approach.
+        For more information look at
+        https://pymatgen.org/pymatgen.symmetry.html#pymatgen.symmetry.bandstructure.HighSymmKpath
+    parallel_info
+        Dictionary containing information about the parallelization of the
+        calculation. See the ASE documentation for more information.
+    test_run
+        If True, a test run is performed to check that the calculation input_data is correct or
+        to generate some files/info if needed.
+    job_params
+        Custom parameters to pass to each Job in the Flow. This is a dictinoary where
+        the keys are the names of the jobs and the values are dictionaries of parameters.
+
+    Returns
+    -------
+    BandsSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][].
+        See the type-hint for the data structure.
+    """
+
+    results = {}
+    bands_kwargs = job_params.get("bands", {})
+
+    bands_result = _bands(
+        atoms=atoms,
+        prev_dir=prev_dir,
+        primitive=primitive,
+        parallel_info=parallel_info,
+        test_run=test_run,
+        **bands_kwargs,
+    )
+
+    if run_bands_pp:
+        bands_pp_kwargs = job_params.get("bands_pp", {})
+        prev_dir = bands_result["dir_name"]
+        bands_pp_results = _bands_pp(
+            atoms, prev_dir, parallel_info, test_run, **bands_pp_kwargs
+        )
+
+    if run_fermi_surface:
+        fermi_kwargs = job_params.get("fermi_surface", {})
+        prev_dir = bands_result["dir_name"]
+        fermi_results = _fermi_surface(
+            atoms, prev_dir, parallel_info, test_run, **fermi_kwargs
+        )
+
+    results["bands"] = bands_result
+    results["bands_pp"] = bands_pp_results if run_bands_pp else None
+    results["fermi_surface"] = fermi_results if run_fermi_surface else None
```

Comment:
Move this into the corresponding `if` statement that way we don't have the entry if the job isn't run.